### PR TITLE
Enable strict compile options on all scala projects and fix warnings.

### PR DESCRIPTION
**Pull request by markusthoemmes**
## Description
Not all scala projects had the compiler flags we usually use so some warnings went under the radar. This fixes this.

## Types of changes

- [X] Bug fix (generally a non-breaking change which closes an issue).

## Checklist:

- [X] I signed an [Apache CLA](https://github.com/apache/openwhisk/blob/master/CONTRIBUTING.md).
- [X] I reviewed the [style guides](https://github.com/apache/openwhisk/wiki/Contributing:-Git-guidelines#code-readiness) and followed the recommendations (Travis CI will check :).

### CODE REVIEW

File: tools/admin/src/main/scala/org/apache/openwhisk/core/database/LimitsCommand.scala [hunk 1]
```diff
@@ -19,26 +19,18 @@ package org.apache.openwhisk.core.database
...
-  WhiskDocumentReader
-}
+import org.apache.openwhisk.core.entity._
 import org.apache.openwhisk.http.Messages
 import org.apache.openwhisk.spi.SpiLoader
+import org.rogach.scallop.{ScallopConfBase, Subcommand}
+import spray.json.{JsObject, JsString, JsValue, RootJsonFormat}
 
 import scala.concurrent.{ExecutionContext, Future}
+import scala.language.reflectiveCalls
```
Comment (markusthoemmes) [inline tools/admin/src/main/scala/org/apache/openwhisk/core/database/LimitsCommand.scala]: This is the important change in both files. We're using structural types here.

File: core/standalone/build.gradle [hunk 2]
```diff
@@ -162,6 +162,10 @@ dependencies {
     testCompile 'org.scalatest:scalatest_2.12:3.0.5'
 }
 
+tasks.withType(ScalaCompile) {
```
Comment (chetanmeh) [inline core/standalone/build.gradle]: Better to move this to `build.gradle` within `subprojects` part such that it always gets applied on all project

```
pluginManager.withPlugin('scala') {
        tasks.withType(ScalaCompile) {
            scalaCompileOptions.additionalParameters = gradle.scala.compileFlags
        }
    }
```

Comment (markusthoemmes) [reply core/standalone/build.gradle]: I do have that on my TODO list to do in a separate PR. I punted on it in this one as it'll touch all scala projects and I wanted to do some more consolidation in that. Shall I throw it all onto this PR still?

Comment (chetanmeh) [reply core/standalone/build.gradle]: Then lets do that in separate PR

**Pull request merged**